ccmlib/node.py

```diff
@@ -651,7 +654,10 @@ def stop(self, wait=True, wait_other_notice=False, gently=True):
 
             still_running = self.is_running()
             if still_running and wait:
-                wait_time_sec = 1
+                # The sum of 7 sleeps starting at 1 and doubling each time
+                # is 2**7-1 (=127). So to sleep an arbitrary wait_seconds
+                # we need the first sleep to be wait_seconds/(2**7-1).
+                wait_time_sec = wait_seconds/(2**7-1.0)
```

Comment:
what happens if wait_seconds passed is 10 ...

will this be rounded down to 0 ... and then there will be no wait ?

Reply:
what happens when its 150 seconds ... will it also work ? or will it be rounded down wait_time_sec to 1 and then we will wait only for 127 seconds ?

Reply:
No, no rounding. Because of the "1.0" I used in the formula, the entire calculation is done in floating point. You can end up with wait_time_sec = 0.1 or 2.3, or whatever. This floating-pointed-ness will "infect" all the other calculations in the code, and sleep of floating point also works correctly.

So I think this code will work correctly for any number, not just a power of two.

Reply:
An example:
```
>>> wait_seconds=10
>>> wait_time_sec = wait_seconds/(2**7-1.0)
>>> print(wait_time_sec)
0.0787401574803
```

Reply:
my concern is with the actual python time.sleep function will it support the fractions ?

time.sleep(wait_time_sec)

if it does then its ok :)

Reply:
It does. I checked. 
By the way, even it didn't, it doesn't really matter - it could only break the first, tiny, sleeps. The last sleeps are anyway very long and the fractional part in them is not important.